test/blackbox-tests/test-cases/install-libdir/run.t

```diff
@@ -13,3 +13,54 @@
   Installing install/lib/foo/foo.dune
   Installing install/lib/foo/foo.ml
   Installing install/lib/foo/opam
+  Installing install/bin/exec
+
+If prefix is passed, the default for libdir is `$prefix/lib`:
+
+  $ dune install --prefix install
+  Installing install/lib/foo/META
+  Installing install/lib/foo/foo$ext_lib
+  Installing install/lib/foo/foo.cma
+  Installing install/lib/foo/foo.cmi
+  Installing install/lib/foo/foo.cmt
+  Installing install/lib/foo/foo.cmx
+  Installing install/lib/foo/foo.cmxa
+  Installing install/lib/foo/foo.cmxs
+  Installing install/lib/foo/foo.dune
+  Installing install/lib/foo/foo.ml
+  Installing install/lib/foo/opam
+  Installing install/bin/exec
+
+If prefix is not passed, libdir defaults to the output of `ocamlfind printconf
+destdir`:
+
+  $ export OCAMLFIND_DESTDIR=$(mktemp -d); dune install 2>&1 | sed s\#$OCAMLFIND_DESTDIR\#OCAMLFIND_DESTDIR\#\;s\#$(opam config var prefix)\#OPAM_VAR_PREFIX\# ; rm -r $OCAMLFIND_DESTDIR
+  Installing OCAMLFIND_DESTDIR/foo/META
+  Installing OCAMLFIND_DESTDIR/foo/foo$ext_lib
+  Installing OCAMLFIND_DESTDIR/foo/foo.cma
+  Installing OCAMLFIND_DESTDIR/foo/foo.cmi
+  Installing OCAMLFIND_DESTDIR/foo/foo.cmt
+  Installing OCAMLFIND_DESTDIR/foo/foo.cmx
+  Installing OCAMLFIND_DESTDIR/foo/foo.cmxa
+  Installing OCAMLFIND_DESTDIR/foo/foo.cmxs
+  Installing OCAMLFIND_DESTDIR/foo/foo.dune
+  Installing OCAMLFIND_DESTDIR/foo/foo.ml
+  Installing OCAMLFIND_DESTDIR/foo/opam
+  Installing OPAM_VAR_PREFIX/bin/exec
+
+If only libdir is passed, binaries are installed under prefix/bin and libraries
+in libdir:
+
+  $ export OCAMLFIND_DESTDIR=$(mktemp -d) LIBDIR=$(mktemp -d); dune install --libdir $LIBDIR 2>&1 | sed s\#$LIBDIR\#LIBDIR\#\;s\#$(opam config var prefix)\#OPAM_VAR_PREFIX\# ; rm -r $OCAMLFIND_DESTDIR $LIBDIR
```

Comment:
Is it not possible to use a local directory for installation? I.e. something like `./install_dest`?

Reply:
If libdir is relative, it will install to `OPAM_VAR_PREFIX/libdir/foo/` instead. I used a global directory so that it avoids installing (even temporarily) in `OPAM_VAR_PREFIX`. But actually this already does it for `bin/`. It doesn't seem great.

What do you think about adding a `--dry-run` option that only displays what would be copied? That way these tests could just use global hardcoded directories.

Reply:
Well you could just turn the relative dir to absolute with something like `$(pwd)/local_dest_dir`. My main concern is that I run these tests quite often so I don't want to pollute my temp dir too much. It's not the most important concern though, so I'm willing to overlook this.

A `--dry-run` would option would be fine as well. It would be useful for other purposes too.

Reply:
They're supposed to be cleaned up, but that's a bit hackish. I opened a PR to have dry-run first.